### DISCUSSION TITLE
tests source code has hardcoded tif file

### DISCUSSION THREAD

**Discussion by kahlenberg** [pelicanmapping/rocky#42] [Q&A / Support] [answered]
Hi,
I have 2 issues about tests source code:

1) In line 308 there is a hardcoded tif file and I don't know where to get these files. My tests are failing (maybe there are multiple hardcoded file names) https://github.com/pelicanmapping/rocky/blob/main/src/tests/tests.cpp#L308

2) I had to clone the repository https://github.com/OSGeo/PROJ-data for other tests. I didn't find it in the documentation. An entry in the documentation should be nice.

**Comment by gwaldron** [comment, marked as answer]
Thanks. I replaced the GDAL test with a WMTS connection string so it won't use a hard-coded TIFF.

It is true that you need PROJ-data to successfully run some of the tests, and the test output tells you that:
```
[rocky info] Note: if you see SRS/VDatum errors, check that you have the NGA grid in your share/proj or PROJ_DATA folder! https://github.com/OSGeo/PROJ-data/blob/master/us_nga/us_nga_egm96_15.tif
```

Getting that file from PROJ-data is only necessary if you plan to use the EGM96 vertical datum. At some point we might consider including it in the rocky distro directly.
